packages/core/src/rating-group/rating-group-root.tsx

```diff
@@ -0,0 +1,223 @@
+} from "./rating-group-context";
+
+export interface RatingGroupRootOptions {
+	/** The current rating value. */
+	value?: number;
+
+	/**
+	 * The initial value of the rating group when it is first rendered.
+	 * Use when you do not need to control the state of the rating group.
+	 */
+	defaultValue?: number;
+
+	/** Event handler called when the value changes. */
+	onChange?: (value: number) => void;
+
+	/** Whether to allow half ratings. */
+	allowHalf?: boolean;
+
+	/** The axis the rating group items should align with. */
+	orientation?: Orientation;
+
+	/**
+	 * A unique identifier for the component.
+	 * The id is used to generate id attributes for nested components.
+	 * If no id prop is provided, a generated id will be used.
+	 */
+	id?: string;
+
+	/**
+	 * The name of the rating group.
+	 * Submitted with its owning form as part of a name/value pair.
+	 */
+	name?: string;
+
+	/** Whether the rating group should display its "valid" or "invalid" visual styling. */
+	validationState?: ValidationState;
+
+	/** Whether the user must select an item before the owning form can be submitted. */
+	required?: boolean;
+
+	/** Whether the rating group is disabled. */
+	disabled?: boolean;
+
+	/** Whether the rating group is read only. */
+	readOnly?: boolean;
+}
+
+export interface RatingGroupRootCommonProps<
+	T extends HTMLElement = HTMLElement,
+> {
+	id: string;
+	ref: T | ((el: T) => void);
+	"aria-labelledby": string | undefined;
+	"aria-describedby": string | undefined;
+	"aria-label"?: string;
+}
+
+export interface RatingGroupRootRenderProps
+	extends RatingGroupRootCommonProps,
+		FormControlDataSet {
+	role: "radiogroup";
+	"aria-invalid": boolean | undefined;
+	"aria-required": boolean | undefined;
+	"aria-disabled": boolean | undefined;
+	"aria-readonly": boolean | undefined;
+	"aria-orientation": Orientation | undefined;
+}
+
+export type RatingGroupRootProps<
+	T extends ValidComponent | HTMLElement = HTMLElement,
+> = RatingGroupRootOptions & Partial<RatingGroupRootCommonProps<ElementOf<T>>>;
+
+export function RatingGroupRoot<T extends ValidComponent = "div">(
+	props: PolymorphicProps<T, RatingGroupRootProps<T>>,
+) {
+	let ref: HTMLElement | undefined;
+
+	const defaultId = `ratinggroup-${createUniqueId()}`;
+
+	const mergedProps = mergeDefaultProps(
+		{
+			id: defaultId,
+			orientation: "horizontal",
+		},
+		props as RatingGroupRootProps,
+	);
+
+	const [local, formControlProps, others] = splitProps(
+		mergedProps,
+		[
+			"ref",
+			"value",
+			"defaultValue",
+			"onChange",
+			"allowHalf",
+			"orientation",
+			"aria-labelledby",
+			"aria-describedby",
+		],
+		FORM_CONTROL_PROP_NAMES,
+	);
+
+	const [items, setItems] = createSignal<CollectionItemWithRef[]>([]);
+	const { DomCollectionProvider } = createDomCollection({
+		items,
+		onItemsChange: setItems,
+	});
+
+	const [hoveredValue, setHoveredValue] = createSignal(-1);
+
+	const [value, setValue] = createControllableSignal<number>({
```

Comment:
Same query -> in know its done similar in other components like Calendar, etc but is it possible to pass these values by user as well?
And will this be useful in RatingGroup, wdyt @jer3m01 ?

Reply:
Same answer as above.